gtdynamics/kinematics/KinematicsSlice.cpp

```diff
@@ -51,6 +51,23 @@ NonlinearFactorGraph Kinematics::graph<Slice>(const Slice& slice,
   return graph;
 }
 
+template <>
+EqualityConstraints Kinematics::constraints<Slice>(const Slice& slice,
+                                              const Robot& robot) const {
+  EqualityConstraints constraints;
+
+  // Constrain kinematics at joints.
+  for (auto&& joint : robot.joints()) {
+    auto constraint_expr = joint->poseConstraint(slice.k);
+    gtsam::Vector6 tolerance = p_->p_cost_model->sigmas();
+    constraints.emplace_shared<VectorExpressionEquality<6>>(constraint_expr,
```

Comment:
Tolerance is constant over loop: take out of the loop

Reply:
Done